### DISCUSSION TITLE
Is that a bug in 'project_points_to_plane' when Parameters setup 'normal=0,1,0'?

### DISCUSSION THREAD

**Discussion by zhuyeqing1982** [pyvista/pyvista#3614] [Q&A]
PolyDataFilters.project_points_to_plane, when Parameters setup 'normal=0,0,1' or 'normal=1,0,0', the plane image is OK. But when parameters setup 'normal=0,1,0', the plane image is broken, just like render error.  thanks for help.
OS: Windows 10 64bit
Graphics: nvidia rtx 3090
driver: 512.15 DCH

**Comment by banesullivan** [comment]
Can you elaborate, share screenshots, and/or share the error of the "broken" scenario? I suspect the data being projected don't fit on the Y-plane.

Here's an example of a uniform data object that can project to any plane:

```py
import pyvista as pv
sphere = pv.Sphere()

p = pv.Plotter(shape=(2,2), notebook=0)

p.subplot(0, 1)
p.add_mesh(sphere.project_points_to_plane(normal=(1,0,0)), show_edges=True)
p.add_mesh(sphere.outline_corners())

p.subplot(1, 0)
p.add_mesh(sphere.project_points_to_plane(normal=(0,1,0)), show_edges=True)
p.add_mesh(sphere.outline_corners())

p.subplot(1, 1)
p.add_mesh(sphere.project_points_to_plane(normal=(0,0,1)), show_edges=True)
p.add_mesh(sphere.outline_corners())

p.subplot(0, 0)
p.add_mesh(sphere, show_edges=True)
p.add_axes()
p.link_views()
p.view_isometric()
p.show()
```
<img width="624" alt="Screen Shot 2022-11-17 at 9 54 29 AM" src="https://user-images.githubusercontent.com/22067021/202508424-94a68d0d-d56d-4af5-8517-7600d431d12c.png">

**Comment by zhuyeqing1982** [comment]
Hi banesullivan, thanks for help, here is the test code:

import pyvista as pv

def mesh_cmp(mesh, name):
    pts = mesh.points
    mesh[name] = pts[:, 1]
    projected = mesh.project_points_to_plane(normal=(0, 1, 0), inplace=False)
    p = pv.Plotter()
    p.add_mesh(mesh, cmap='viridis', lighting=False)
    p.add_mesh(projected, cmap='viridis', lighting=False)
    p.show()

if __name__ == '__main__':
    mesh = pv.read('pcldata/arm2.obj')
    mesh_cmp(mesh, 'prick_test')

Here is the "broken" scenario：
![image](https://user-images.githubusercontent.com/118452633/202587029-3a1d99e3-392c-4d42-a80b-e640db988795.png)
![image](https://user-images.githubusercontent.com/118452633/202587092-68cbd58a-9463-4b05-9fa6-a1e903423515.png)

**Reply by zhuyeqing1982** [reply]
Ohhhhhh......It's the same problem when I use pv.Sphere() with colour map.


`import pyvista as pv
def mesh_cmp(mesh, name):
    pts = mesh.points
    mesh[name] = pts[:, 1]
    projected = mesh.project_points_to_plane(normal=(0, 1, 0), inplace=False)
    p = pv.Plotter()
    # p.add_axes()
    p.add_mesh(mesh, cmap='fire', lighting=False)
    p.add_mesh(projected, cmap='fire', lighting=False)
    p.show()

if __name__ == '__main__':
    mesh=pv.Sphere()
    mesh_cmp(mesh, 'prick_test')`

![image](https://user-images.githubusercontent.com/118452633/202602208-cd3829b7-6635-4174-ac8f-8c21ce0158e1.png)

**Reply by zhuyeqing1982** [reply]
The problem may happen with cmap, it's seens ok without cmap.
![image](https://user-images.githubusercontent.com/118452633/202698390-ec76d2c9-f407-44c2-bc04-1f3aa90469ff.png)
![image](https://user-images.githubusercontent.com/118452633/202698532-cbae02ea-6916-4cec-a4be-7c84f37c6385.png)
